src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/types.ts

```diff
@@ -41,17 +41,46 @@ interface PublishesOptions {
   totalCardinality$: PublishingSubject<number>;
 }
 
-export type OptionsListComponentApi = OptionsListControlApi &
+type OptionsListState = Pick<DefaultDataControlState, 'fieldName'> &
```

Comment:
It would be ideal if we could revert all the changes in this file now that we're just pre-filling all of required parts of the API in the ESQL control.